### DISCUSSION TITLE
Unable to find repo in ancestor path in non-existent parent path.

### DISCUSSION THREAD

**Discussion by 0xPoe** [libgit2/libgit2#6380] [Q&A] [answered]
Cargo uses [this bind API ](https://docs.rs/git2/latest/src/git2/repo.rs.html#245-260) to find the root repo when users call `cargo new`.
But we found that if it is unable to find repo in ancestor path in non-existent parent path.

For example:
1. init git in `/User/rustin/test`
2. try find the repo in `/User/rustin/test/non-exist` path.

See more at https://github.com/rust-lang/cargo/issues/10981.

Also, you can check out https://github.com/rust-lang/cargo/pull/10987 to see if you pass an existing path for it. It will work.

Please correct me if I have misunderstood the meaning of this API. Thank you!

**Comment by ethomson** [comment, marked as answer]
This is definitely by design. The goal of this api is to discover the root of the repo for your current working directory. (So that, eg, `git add .` works several levels down.)

Not necessarily opposed to adding an option here that the trailing paths may be imaginary. We could convert `across_fs` to flags with little impact on compatibility.

**Reply by 0xPoe** [reply]
> Not necessarily opposed to adding an option here that the trailing paths may be imaginary. We could convert `across_fs` to flags with little impact on compatibility.

Thanks for your help!
Should I open a feature request issue or we can just convert this discussion to an issue?
